test/bigml/test/api/source.clj

```diff
@@ -8,11 +8,13 @@
   (:use clojure.test))
 
 (defn- create-and-test [artifact & params]
-  (let [initial (api/get-final (apply source/create artifact params))
+  (let [{:keys [username api_key dev_mode] :as sticky-pars} params
```

Comment:
thanks @ashenfad. but let me just check my understanding of things a little bit...
My reasoning was this: `params` can be used to pass any source creation options; on the other hand, we only want to pass auth credentials + `dev_mode` to subsequent calls. That is why I introduced `sticky-params` the way I did. It is true that we have currently no unit tests that use creation parameters, but I think this will be the case as we grow the test suite. Does it make more sense?